python/dynamic_graph/sot/torque_control/tests/test_torque_offset_estimator.py

```diff
@@ -0,0 +1,169 @@
+taskLH.opmodif = matrixToTuple(handMgrip)
+taskLH.feature.frame('desired')
+
+# --- STATIC COM (if not walking)
+taskCom = MetaTaskKineCom(robot.dynamic)
+robot.dynamic.com.recompute(0)
+taskCom.featureDes.errorIN.value = robot.dynamic.com.value
+taskCom.task.controlGain.value = 10
+
+
+# --- CONTACTS
+#define contactLF and contactRF
+for name,joint in [ ['LF',robot.OperationalPointsMap['left-ankle']], ['RF',robot.OperationalPointsMap['right-ankle'] ] ]:
+    contact = MetaTaskKine6d('contact'+name,robot.dynamic,name,joint)
+    contact.feature.frame('desired')
+    contact.gain.setConstant(10)
+    contact.keep()
+    locals()['contact'+name] = contact
+
+
+targetRH = (0.5,-0.2,1.0)
+targetLH = (0.5,0.2,1.0)
+
+#addRobotViewer(robot, small=False)
+#robot.viewer.updateElementConfig('zmp',target+(0,0,0))
+
+gotoNd(taskRH,targetRH,'111',(4.9,0.9,0.01,0.9))
+gotoNd(taskLH,targetLH,'111',(4.9,0.9,0.01,0.9))
+sot.push(contactRF.task.name)
+sot.push(contactLF.task.name)
+sot.push(taskCom.task.name)
+sot.push(taskLH.task.name)
+sot.push(taskRH.task.name)
+
+
+# Sensor Calibration Entity.
+from dynamic_graph.sot.torque_control.torque_offset_estimator import TorqueOffsetEstimator as TOE
+import numpy as np
+
+toe = TOE("test_entity_toe")
+id4 = ((1.0,0.0,0.0,0.0),(0.0,1.0,0.0,0.0),(0.0,0.0,1.0,0.0),(0.0,0.0,0.0,1.0))
+toe.init(urdfPath, id4, 0.1)
+plug(robot.device.state, toe.base6d_encoders)
+toe.gyroscope.value = (0.,0.,0.)
+
+from pinocchio import Motion
+gravity = Motion.Zero()
+gravity.linear = np.asarray((0.0,0.0,-9.81))
+njoints = pinocchioRobot.model.nv-6
+
+#-------------------------------------------------------------------------------
+#----- MAIN LOOP ---------------------------------------------------------------
+#-------------------------------------------------------------------------------
+robot.device.increment(dt)
+from pinocchio import rnea
+
+np.set_printoptions(suppress=True)
+
+def runner(n):
+    for i in xrange(n):
+        q_pin = fromSotToPinocchio(robot.device.state.value)
+        nvZero = np.asarray((0.0,)*pinocchioRobot.model.nv)
+        tau = np.asarray(rnea(pinocchioRobot.model, pinocchioRobot.data, q_pin, nvZero, nvZero)).squeeze()
+        tau_offset = np.asarray((0.1,)*njoints).squeeze()
+        toe.jointTorques.value = tau_offset + tau[6:]
+        pinocchioRobot.forwardKinematics(q_pin)
+        toe.accelerometer.value = np.asarray((pinocchioRobot.data.oMi[15].inverse()*pinocchioRobot.data.oMi[1]*gravity).linear).squeeze()
+
+        toe.jointTorquesEstimated.recompute(robot.device.state.time)
+        #print toe.jointTorques.value
+        entity_offset = np.asarray(toe.jointTorquesEstimated.value).squeeze() - np.asarray(toe.jointTorques.value).squeeze()
+
+        robot.device.increment(dt)
+        pinocchioRobot.display(q_pin)
+
+
+runner(10)
+toe.computeOffset(100,1.0)
+runner(100)
+runner(1000)
+
+print toe.getSensorOffsets()
```

Comment:
Maybe here you wanna print something that is more self-explaining, such as printing the expected torque offset, followed by the estimated sensor offset, and finally assert that the norm of the difference between the two is smaller than 1e-6.